### PR TITLE
Extend moves on a cdb PV line

### PR DESCRIPTION
**Pull request by vondele**
probe cdb for the PV of the position searched, and extend the bestmove on positions on the PV, in case the depth is below the depth remaining the reach the PV end.

### CODE REVIEW

**Comment by robertnurnberg** [comment]
I only glanced at this, will check later in more detail. But looks good.

A couple of thoughts:
* rename `cdbpv` to `cdbpvLength` ?
* rename `pvextensions` to `get_cdbpvlength` ?

Also, how about adding to the output `Search at depth  2` also something like seldepth/selective depth that accounts for the value in `cdbpv` for the root pos? And move the print statement right after the call to `pvextensions`, before search starts? This would warn users that they may have to wait longer than expected for the search to finish.... (Not sure if the name seldepth makes any sense here.)

**Comment by vondele** [comment]
Made the changes as suggested above, with somewhat different names.

**Comment by vondele** [comment]
so, have been playing with this, this works very well, IMO.

**Comment by robertnurnberg** [comment]
Agreed, this is a very nice addition.

One question: at present the leaf node in the cdb PV gets a `cdbPvToLeaf` value of 0, meaning that this position itself will not get a search extension (through the new criterion at least). Is this by design? If not, then changing L358 to `if newdepth <= self.cdbPvToLeaf[epd]:` would change that behaviour. It would also be more generous to the other nodes on the PV line, not sure if that is something you'd want.

So I think this is ready to merge. I have my usual inconsequential clean-ups, which I can do in a separate PR once this one is merged. :)

**Comment by vondele** [comment]
I played a bit with that criterion on L358, we don't want to extend too much, as this especially makes depth 1 more expensive, and is relatively inconsequential later on.

File: cdbsearch.py [hunk 1]
```diff
@@ -100,6 +103,32 @@ def __apicall(self, url, timeout):
...
+            pvlen = remaining = len(pv)
+            board = chess.Board(epd)
+            self.cdbPvToLeaf[board.epd()] = remaining
+            self.executorWork.submit(self.queryall, board.epd())
+            for m in pv:
+                move = chess.Move.from_uci(m)
+                board.push(move)
+                remaining -= 1
+                self.cdbPvToLeaf[board.epd()] = remaining
+                self.executorWork.submit(self.queryall, board.epd())
```
Comment (robertnurnberg) [inline cdbsearch.py:129]: So in my cleanup I stumbled upon this line: The skipTT argument here seems to be missing. My plan is to make skipTT=False the default, but I do not understand how the code could run as it is? Does `submit` populate missing arguments with zeros or something? In that case, the code would work as intended.

Comment (vondele) [reply cdbsearch.py:129]: I agree this needs fixing, even if there is some default behavior. I suspect it might be reading some random data.

**Pull request merged**